### DISCUSSION TITLE
Where can I find the up-to-date source of protogen-php-grpc plugin?

### DISCUSSION THREAD

**Discussion by kindacommander** [roadrunner-server/roadrunner#993] [Q&A] [answered]
Is there a way to build protogen-php-grpc plugin from source instead of downloading pre-built binary? I've tried building the latest version from [php-repo](https://github.com/spiral/php-grpc), but it's definitely outdated. Where can i find the source code?

**Comment by rustatian** [comment, marked as answer]
Hey @kindacommander. [php-repo](https://github.com/spiral/php-grpc) was deprecated. You can download `protoc-gen-php-grpc` plugin directly from the `roadrunner` releases page: [click](https://github.com/roadrunner-server/roadrunner/releases).

**Reply by kindacommander** [reply]
Yeah, i got it. But what about the source code?

**Reply by programmador** [reply]
Same question. Strange that binaries are present on releases page of this repo and the source code is missing in the repo at the same time.
Maybe it was moved somewhere... Anyway it's missing in the `cmd` directory where it was present in the abovementioned archived repo.

**Reply by rustatian** [reply]
> Yeah, i got it. But what about the source code?

If you have a look at the deprecated repo, you'll see, that the RRv2 sources were moved here: https://github.com/spiral/roadrunner-grpc
![image](https://user-images.githubusercontent.com/8040338/153034316-1c9ff02a-f3ec-47e8-bd64-e9535cb6a6bc.png)

**Reply by programmador** [reply]
The link leads to current repository. Which contains protoc plugin releases (binary), not it's sources. So where are they?

**Reply by rustatian** [reply]
What sources are you talking about?

**Reply by programmador** [reply]
sources of protoc plugin

**Reply by rustatian** [reply]
Golang plugin you can found here: https://github.com/roadrunner-server/grpc

**Reply by programmador** [reply]
```
> grep -irn 'protoc-gen-php-grpc'
.github/workflows/release_grpc.yml:58:          echo "::set-output name=binary-name::protoc-gen-php-grpc`[ ${{ matrix.os }} = 'windows' ] && echo '.exe'`"
.github/workflows/release_grpc.yml:69:          go build -trimpath -ldflags "$LDFLAGS" -o "./${{ steps.values.outputs.binary-name }}" protoc_plugins/protoc-gen-php-grpc/main.go
.github/workflows/release_grpc.yml:75:          echo "::set-output name=name::protoc-gen-php-grpc-${{ steps.values.outputs.version }}-$(
> find -iname '*protoc_plugins*'
> find -iname 'main.go'
./cmd/rr/main.go
> find -iname '*gen*'
>
```

There is a mention about it in github CI scripts but it's not clear what are the binaries being built from.
Seems like no building documentation exists either.

**Reply by programmador** [reply]
Oh, thank You so much! It seems like it is the source we're looking for.

**Reply by kindacommander** [reply]
Thanks, @rustatian, that's exactly what I was looking for.

**Reply by rustatian** [reply]
> Seems like no building documentation exists either.

We don't provide build-from-source documentation for the `protoc-gen-php-grpc` plugin, because this is not a part of the public API. Sources for this plugin might be changed at any moment. The only source of truth here - is binaries on the releases page.
